### PR TITLE
ci: Update Codeowners with data platform paths

### LINKED ISSUE #26068: [Task]: Update CODEOWNERS file for Data platform
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- identify the code paths which fall under data platform both from server and client side
- add them to codeowners file

### PR DESCRIPTION
**Pull request by riteshkew**
#### PR fixes following issue(s)
Fixes #26068 
> 
>
>
- Chore (housekeeping or task changes that don't impact user perception)
>
>
>
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -126,4 +126,72 @@ app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServ
...
+app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/OAuth2ResponseDTO.java @nidhi-nair @sondermanish @tkappsmith
+
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/ActionControllerCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/Datasource* @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/PageControllerCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ActionCollectionDTO.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ActionDTO.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithErrorCode.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/PermissionUtils.java @nidhi-nair @sondermanish @tkappsmith
```
Comment (nidhi-nair) [inline CODEOWNERS:177]: This one's global tbh, should we limit this list to svc, ctrlr, repo, soln, model and dto files?

File: CODEOWNERS [hunk 2]
```diff
@@ -126,4 +126,72 @@ app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServ
...
+app/client/src/transformers/RestAPIDatasourceFormTransformer.ts @ayushpahwa @ChandanBalajiBP
+
+
+app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ActionExecutionResult.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/Datasource* @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/OAuth2ResponseDTO.java @nidhi-nair @sondermanish @tkappsmith
+
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/ActionControllerCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/Datasource* @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/PageControllerCE.java @nidhi-nair @sondermanish @tkappsmith
```
Comment (nidhi-nair) [inline CODEOWNERS:172]: Why this one?

File: CODEOWNERS [hunk 3]
```diff
@@ -126,4 +126,72 @@ app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServ
...
+app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ActionCollectionDTO.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ActionDTO.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithErrorCode.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/PermissionUtils.java @nidhi-nair @sondermanish @tkappsmith
+
+app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/*Datasource* @nidhi-nair @sondermanish @tkappsmith
+
+app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/*Datasource* @nidhi-nair @sondermanish @tkappsmith
```
Comment (nidhi-nair) [inline CODEOWNERS:181]: I like this kinda usage, let's use it for all?

*Action* and *Datasource* under all folders?

File: CODEOWNERS [hunk 4]
```diff
@@ -126,4 +126,72 @@ app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServ
...
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/ActionControllerCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/Datasource* @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/PageControllerCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ActionCollectionDTO.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ActionDTO.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithErrorCode.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/PermissionUtils.java @nidhi-nair @sondermanish @tkappsmith
+
+app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/*Datasource* @nidhi-nair @sondermanish @tkappsmith
```
Comment (sondermanish) [inline CODEOWNERS:179]: we also require ..../*Environment* for repositories.

File: CODEOWNERS [hunk 5]
```diff
@@ -126,4 +126,72 @@ app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServ
...
+
+app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/*Datasource* @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCEImpl.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/AuthenticationValidatorCE.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/AuthenticationValidatorCEImpl.java @nidhi-nair @sondermanish @tkappsmith
+app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/LayoutActionServiceCEImpl.java @nidhi-nair @sondermanish @tkappsmith
```
Comment (sondermanish) [inline CODEOWNERS:186]: We can have similar usages [*Datasource*, *Environment*] for services and solutions as well

File: CODEOWNERS (hunk 4 above)
Comment (riteshkew) [reply CODEOWNERS]: @sondermanish I couldn't find any filename with Environment. Can you give an example for the file you are talking about.

File: CODEOWNERS (hunk 5 above)
Comment (riteshkew) [reply CODEOWNERS]: Done.

File: CODEOWNERS (hunk 4 above)
Comment (nidhi-nair) [reply CODEOWNERS]: These would be on EE repo I guess.

**Review by nidhi-nair — APPROVED** [review]
There's a few other features like refactoring, datatypes, base plugin interfaces and implementations, constants, spans, possibly even chat service that we may want to include here. But may be we can take those as a follow up? Based on how effective this is?

File: CODEOWNERS (hunk 4 above)
Comment (riteshkew) [reply CODEOWNERS]: I'll follow up with EE repo codepaths next.

**Pull request merged**